src/store/reducers.tsx

```diff
@@ -38,35 +31,63 @@ const card = (state: any = {}, action: any) => {
   }
 };
 
+/*
+  Cards are:
+    A UUID identifier
+    Text of the card body
+    What round they were selected in
+*/
+
 const initialCards = CARD_TEXTS.map((text: any) => ({
   key: uuid.v4(),
   text,
-  mark: false,
-  discard: false
+  mark: 0,
 }));
 
-const cards = (state: any = initialCards, action: any) => {
-  switch (action.type) {
-    case 'MARK_CARD':
-    case 'TOGGLE_CARD':
-      return state.map((c: any) => card(c, action));
-    default:
-      return state;
+const computeCardsNeeded = (cards: any) => {
+  const arr = new Array<number>();
+  arr.push(0);
+  let remaining = cards.length
+  while (remaining >= 1) {
+    remaining = Math.floor(remaining/2);
+    arr.push(remaining);
   }
-};
+  return arr;
+}
+const initialState = {
+  cards: initialCards,
+  stage: 1,
+  cardsNeeded: computeCardsNeeded(initialCards)
+}
 
-const stage = (state: any = 1, action: any) => {
+const reducer = (state: any = initialState, action: any) => {
```

Comment:
Since I refactored to only a single reducer, I thought that calling it "reducer" was a common Reduxism